go/connector/run.go

```diff
@@ -209,22 +288,25 @@ func runCommand(
 		}
 	}(cmd.Process.Signal)
 
-	err = cmd.Wait()
-	var closeErr = cmd.Stdout.(io.Closer).Close()
-	// Ignore error on closing stderr because it's already logged by the forwarder
+	var waitErr = cmd.Wait()
+	if waitErr != nil {
+		if ctx.Err() == nil {
+			// Expect a clean exit if the context wasn't cancelled.
+			// Log the raw error, since we've already logged everything that was printed to stderr.
+			logger.Log(logrus.ErrorLevel, logrus.Fields{"error": waitErr}, "connector failed")
+			fe.onError(fmt.Errorf("go.estuary.dev/E116: connector failed, with error: %w\nwith stderr:\n\n%s",
+				waitErr, cmd.Stderr.(*connectorStderr).buffer.String()))
+		} else {
+			fe.onError(ctx.Err())
+		}
+	}
 	_ = stderrForwarder.Close()
 
-	if err == nil {
-		// Expect clean output after a clean exit, regardless of cancellation status.
-		fe.onError(closeErr)
-	} else if ctx.Err() == nil {
-		// Expect a clean exit if the context wasn't cancelled.
-		// Log the raw error, since we've already logged everything that was printed to stderr.
-		logger.Log(logrus.ErrorLevel, logrus.Fields{"error": err}, "connector failed")
-		fe.onError(fmt.Errorf("go.estuary.dev/E116: connector failed, with error: %w\nwith stderr:\n\n%s",
-			err, cmd.Stderr.(*connectorStderr).buffer.String()))
-	} else {
-		fe.onError(ctx.Err())
+	if port != "" {
+		var closeErr = conn.Close()
```

Comment:
Is there a case where `conn` is `nil` here where the connector is using a port but the connection can't be established within the timeout above?